libutils/string_lib.c

```diff
@@ -803,26 +803,26 @@ bool ReplaceStr(const char *in, char *out, int outSz, const char *from, const ch
 /**
  * Replace all occurrences of #find with #replace.
  *
- * @return the length of #buf or (size_t) -1 in case of overflow, or 0
- *         if no replace took place.
+ * @return the length of #buf or -1 in case of overflow, or 0 if no replace
+ *         took place.
  */
-size_t StringReplace(char *buf, size_t buf_size,
-                     const char *find, const char *replace)
+ssize_t StringReplace(char *buf, size_t buf_size,
+                      const char *find, const char *replace)
 {
     assert(find[0] != '\0');
 
     char *p = strstr(buf, find);
     if (p == NULL)
     {
-        return 0;
+        return strlen(buf);
```

Comment:
I think you are changing the *0 if no replace took place* part of the behavior here.

Reply:
That was intentional. I think it's a more consistent behaviour in the case that you are using the result of the function as a way to gauge how long it is. But I also see that knowing if there was some replacement would be a good idea.

Should I change it back, or does my explanation make sense?

Reply:
Never mind. I'm being silly.